### PR TITLE
Integration specs n1 problem

### PR DESCRIPTION
**Pull request by Berkay-akbas**
# Hello Code Reviewer 👋 

# Changes ♻️ 
- Integration specs for User Index, User Show, User Post Index, and User Post Show are implemented with a total of 29 tests.
- Solved the n+1 problem which was on post show page.

Thanks for reviewing our code!

### CODE REVIEW

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @Berkay-akbas,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Descriptive pull request message ✔️
- No linter errors ✔️
- All you tests are passing ✔️
almoo

## Required Changes ♻️
- [ ] Let's solves N+1 problem when fetching all posts and their comments for a user
- _Hint_: Example in the posts#index actions students should use @posts = @user.posts.includes(:comments) instead of @posts = @user.posts

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -9,7 +11,7 @@
 
 
 <% @count = 0 %>
-<% @user.posts.each do |post|  %>
+<% @user.posts.includes([:comments]).each do |post|  %>
```
Comment (Berkay-akbas) [inline app/views/posts/index.html.erb:14]: @uwadonat Hello and thank you for the review, Here you can see it in the view file for posts index. It was one of the changes.

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @Berkay-akbas, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
Hi @Berkay-akbas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**